bldsys/cmake/sample_helper.cmake

```diff
@@ -17,159 +17,57 @@
-        list(APPEND SHADER_FILES_GLSL ${TARGET_SHADER_FILES_GLSL})
-        foreach(SHADER_FILE_GLSL ${SHADER_FILES_GLSL})
-            list(APPEND SHADERS_GLSL "${PROJECT_SOURCE_DIR}/shaders/${SHADER_FILE_GLSL}")
-        endforeach()        
-    endif()
-
-    add_project(
-        TYPE "Sample"
-        ID ${TARGET_ID}
-        CATEGORY ${TARGET_CATEGORY}
-        AUTHOR ${TARGET_AUTHOR}
-        NAME ${TARGET_NAME}
-        DESCRIPTION ${TARGET_DESCRIPTION}
-        TAGS 
-            ${TARGET_TAGS}
-        FILES
-            ${SRC_FILES}
-        LIBS
-            ${TARGET_LIBS}
-        SHADERS_GLSL
-            ${SHADERS_GLSL})
+add_custom_target(vkb__samples)
 
-endfunction()
-
-function(vkb_add_test)
+function(vkb__register_sample)
     set(options)
-    set(oneValueArgs ID)
-    set(multiValueArgs)
+    set(oneValueArgs NAME)
+    set(multiValueArgs SRC LINK_LIBS INCLUDE_DIRS)
 
     cmake_parse_arguments(TARGET "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
-    add_project(
-        TYPE "Test"
-        ID ${TARGET_ID}
-        CATEGORY "Tests"
-        AUTHOR " "
-        NAME ${TARGET_ID}
-        DESCRIPTION " "
-        VENDOR_TAG " "
-        LIBS test_framework
-        FILES
-            ${CMAKE_CURRENT_SOURCE_DIR}/${TARGET_ID}.h
-            ${CMAKE_CURRENT_SOURCE_DIR}/${TARGET_ID}.cpp)
-endfunction()
-
-function(add_project)
-    set(options)  
-    set(oneValueArgs TYPE ID CATEGORY AUTHOR NAME DESCRIPTION)
-    set(multiValueArgs TAGS FILES LIBS SHADERS_GLSL)
-
-    cmake_parse_arguments(TARGET "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
-
-    if(${TARGET_TYPE} STREQUAL "Sample")
-        set("VKB_${TARGET_ID}" ON CACHE BOOL "Build sample ${TARGET_ID}")
+    if(TARGET_NAME STREQUAL "")
+        message(FATAL_ERROR "NAME must be defined in vkb__register_tests")
     endif()
 
-    if(NOT ${VKB_${TARGET_ID}})
-        message(STATUS "${TARGET_TYPE} `${TARGET_ID}` - DISABLED")
-        return()
-    endif()
+    if(TARGET_SRC) ## Create static library
+        message("ADDING SAMPLE: sample__${TARGET_NAME}")
 
-    message(STATUS "${TARGET_TYPE} `${TARGET_ID}` - BUILD")
+        add_library("sample__${TARGET_NAME}" SHARED ${TARGET_SRC})
 
-    # create project (object target - reused by app target)
-    project(${TARGET_ID} LANGUAGES C CXX)
+        target_link_libraries("sample__${TARGET_NAME}" PUBLIC vkb__platform_headers) # sample_main
 
-    source_group("\\" FILES ${TARGET_FILES})
+        if(TARGET_LINK_LIBS)
+            target_link_libraries("sample__${TARGET_NAME}" PUBLIC ${TARGET_LINK_LIBS})
+        endif()
 
-    # Add shaders to project group
-    if (SHADERS_GLSL)
-        source_group("\\Shaders" FILES ${SHADERS_GLSL})
+        if(TARGET_INCLUDE_DIRS)
+            target_include_directories("sample__${TARGET_NAME}" PUBLIC ${TARGET_INCLUDE_DIRS})
+        endif()
+        
+        if(MSVC)
+            target_compile_options("sample__${TARGET_NAME}" PRIVATE /W4 /WX)
```

Comment:
Dunno if you get any warnings in MSVC when compiling with this due to /W3 and /Md being set automatically D9025 might crop up.  However, just in case it happens:
https://cmake.org/cmake/help/v3.15/policy/CMP0092.html
https://cmake.org/cmake/help/v3.15/policy/CMP0091.html
add:
if (POLICY CMP0091)
  cmake_policy(SET CMP0091 NEW)
endif ()
if (POLICY CMP0092)
  cmake_policy(SET CMP0092 NEW)
endif ()